### ISSUE TITLE
Studio Controls 2.1.4 breaking system?

### ISSUE THREAD

**Issue by Krauus** [ovenwerks/studio-controls#51]
Host OS: Linux Mint 20.10 XFCE
Ubuntu Studio Installer: All options selected except for Wallpapers. **Backport PPA Enabled**
Studio Controls: version 2.1.4

### The issue:

My entire Studio Control system is broken at system boot-up. I don't know what caused it. I'm not even sure where to begin.

My best guess is things went wrong when I enabled the backports PPA in order to have the most newest version of Studio Controls so I could aid with the GTK UI design of Studio Controls.

### I'll start by describing my usual Ubuntu Studio setup:

Scarlett 18i20 USB Audio Interface connected via USB 2.0
Channel 3 Input: EV-320 Dynamic Mic through a DBX 286s Mic Pre-Amp/Processor Channel Strip

**Pulse Bridges:**
source: "pulse_in" ; channels: 4
source: "discord_in" ; channels: 1
sink: "pulse_out" ; channels 4
sink: "discord_out" ; channels: 2

**Pre-loaded Carla profile**, invoked at login through a bash script in XFCE's "Session and Startup"

```
#!/bin/sh

jack_wait -w && sleep 2 && carla -n ~/carla-profile_01.carxp; bash
```

I've confirmed the script is active by reviewing my running processes.

![image](https://user-images.githubusercontent.com/69413775/111420668-dfa9b180-86a8-11eb-837f-0b803dc0b48b.png)

It is not functioning though, because when I go to my Carla patchbay, the entire configuration is mangled:

![image](https://user-images.githubusercontent.com/69413775/111418943-cc491700-86a5-11eb-94a1-212ecc843e7e.png)

LSP Gate Mono is not supposed to have "Carla.01" in the name. It also lacks any connections from the audio interface or pulse_in bridge.

So, to fix this, I close Carla, then I go into Studio Controls (2.1.4) and click "Start or Restart  JACK"

Instead of restarting, the button causes Jack to stop completely. I click the button again, and Jack starts up.

The pre-loaded Carla bash script process is killed.

I re-open Carla, and re-Open my profile, and it's still mangled, but the connections are back in place:

![image](https://user-images.githubusercontent.com/69413775/111416927-16c89480-86a2-11eb-8a86-316af07d2b71.png)

I'll re-arrange them manually to see things clearly:

![image](https://user-images.githubusercontent.com/69413775/111417000-38298080-86a2-11eb-89e4-d72cf726b53c.png)

Then I'll hit "Save." This usually saves the arrangement, but now that is broken.

Since the pre-loaded script is killed, if I close Carla all my patchbay connections break. So let's try another way...

I restart my computer, restart JACK through the controls, and execute the bash script. It mangles my patchbay again.

How about another way?

If I try manually loading my Carla profile in CLI, here's what happens:

```
$ carla -n ~/carla-profile_01.carxp
Carla 2.2.0 started, status:
  Python version: 3.8.5
  Qt version:     5.12.8
  PyQt version:   5.14.1
  Binary dir:     /usr/lib/carla
  Resources dir:  /usr/share/carla/resources
Carla assertion failure: "value <= 0 || value >= 1024" in file CarlaStandalone.cpp, line 732
Carla assertion failure: "value <= 0 || value >= 1024" in file CarlaStandalone.cpp, line 737
libjack.so.0 loaded successfully!
Carla ready!
```

Things get crazier if I logout / login instead of restarting...

![image](https://user-images.githubusercontent.com/69413775/111421071-973ec380-86a9-11eb-86b1-be80e6a2ad79.png)

Dual bash scripts?

![image](https://user-images.githubusercontent.com/69413775/111421124-ac1b5700-86a9-11eb-9e67-50faab5bcbb5.png)

Wha!? Jack is still running, but it's not even being detected? It's using Pulse instead! Absolutely crazy!

**Comment by ovenwerks** [comment]
"Pre-loaded Carla profile, invoked at login through a bash script in XFCE's "Session and Startup"

#!/bin/sh

jack_wait -w && sleep 2 && carla -n ~/carla-profile_01.carxp; bash"

A) maybe try sleep 10 first. It looks like the Carla profile is being loaded before the bridges are available. sleep 10 is probably (hopefully) longer than needed but may help anyway. This should take care of  the effect in Carla not getting connected.

B) "My best guess is things went wrong when I enabled the backports PPA" As many of your problems are with Carla and not studio controls, it would appear that the upgrade from Carla 2.1 to Carla 2.2 may have caused that problem. What I do is to enable the PPA, Install the package I want from it, then turn the PPA off if I do not want everything. The installer GUI also allows unchecking thing you do not want installed. In any case I would suggest down grading Carla to version 2.1 if that works better for you.

NOTE: The Carla version 2.2 in ubuntu is patched and not the same as Carla 2.2 upstream. It has been patched to avoid causing Jackd to hang/lock up under certain conditions. This may be part of what is causing your problem. The ubuntu team felt that having Carla not work quite as intended was better than having jackd freeze and require a reboot as restarting jackd did not solve the problem. In any case studio controls cannot fix Carla.

C) While you do not say so here, on IRC you mention that you have tried to logout and then login. Please be aware that for all DEs I have tried, this does not work. It appears that no DE does this right. All DEs now use systemd/logind to start their session and for shutdown or reboot, the correctly call a systemd --system command to do this which ends the logind session and shuts down properly. However, because this group of commands is all SU, there is no systemd --system command to logout correctly and the DEs expect than when they stop their session manager this will just happen on it's own. This does not happen and in fact there is a systemd -user command that does do this correctly that should be run as the user rather than root that none of them use. Gnome in fact tries all kinds of wierd stuff to make this work and still fails. I am pretty sure this is the cause of finding things labeled "Carla.01".  The result is that processes started by autostart in the DE are not terminated and the logind session continues to run meaning you have old things running from before in the old session. Reboot instead of logout/login

D) If jackd has been restarted, Carla 2.1 will freeze unless the audio engine is stopped before jackd stops. Carla 2.2 should put a dialog up telling you you need to restart the audio engine before continuing. That is it is possible to restart Carla2.2's audio engine without restarting the application. However, I do not know if it will automatically also reload the profile. As you are having trouble with Carla 2.2 and may go back to Carla 2.1, beware of these things.

E) what can be done in Studio Controls to help? (not fix everything) Studio Controls 2.2 will have the ability to run a script after jack startup and before jack shutdown so that your Carla session is always new renewed if jack is restarted by Studio Controls.

**Comment by Krauus** [comment]
I guess what I need to have is guidance on how to revert things back to their non backport state so that my system works properly, then wait for the devs of these various projects to fix things so that they work. How can I undo all the changes that the "Enable Backport PPA" caused?
